app/controllers/samples_controller.rb

```diff
@@ -1352,8 +1352,33 @@ def taxa_with_reads_suggestions
       return
     end
 
-    taxon_list = taxon_search(query, ["species", "genus"], samples: samples)
+    taxon_list = taxon_search(query, ["species", "genus"])
```

Comment:
This slightly optimizes the reads endpoint.

It turns I don't need the `samples: samples` filter here, since when I calculate the counts, I make a very similar SQL query. Removing the filter removes a redundant SQL query.

Reply:
Do you filter by taxon counts too?
I believe that the samples filter constrains by taxon_counts, so you will not see tax ids that have contigs but not taxon counts.
If you remove the filter you will see all contigs (even if there are no taxon counts). Is this what you want (could be confusing for the end user)?

Reply:
I filter by taxon counts for the taxon count endpoint and contigs for the contig endpoint. I think seeing all contigs even if there are no taxon counts is fine, since the user is downloading contigs, not reads. I think it would be wrong to hide a taxon with contigs but no taxon counts. (although I think this is very rare)

Reply:
The prob is they do not see these contigs anywhere else, which seems to be confusing to the user. 
I guess the long term solution is that we need to show them somewhere in the report.

Reply:
Ahh I see what you mean. The long-term solution sounds good.